### ISSUE TITLE
ZipFS: improper unpacking  of  (XML) files

### ISSUE THREAD

**Issue by zopyx** [PyFilesystem/pyfilesystem2#108]
I have this sample code what will read a ZIP file containing a valid `sd.xml` XML file:

```

import fs.opener

zf, _ = fs.opener.open('zip://sample.zip', writeable=False)
print zf.listdir(u".")
data = zf.open(u'sd.xml', 'rb').read()
print data
```

<img width="1178" alt="bildschirmfoto 2017-11-13 um 19 25 32" src="https://user-images.githubusercontent.com/594239/32742028-7c4b82ce-c8a8-11e7-8635-794fe3ffc12e.png">

The screenshot shows that the closing tag `<smashdocc` is corrupted instead of `<smashdoc>`.

The second screenshot

<img width="1179" alt="bildschirmfoto 2017-11-13 um 19 27 55" src="https://user-images.githubusercontent.com/594239/32742149-cde5b2bc-c8a8-11e7-8642-0386d4872f37.png">

shows that the `sd.xml` XML file is correct.

The related `sample.zip` can be found here: https://public.zopyx.com/sample.zip

**Comment by willmcgugan** [comment]
Possibly an out by one error. Will take a look.

**Comment by althonos** [comment]
Probably my fault, sorry.

**Comment by willmcgugan** [comment]
Fixed in 2.0.17 Thanks to @althonos
